### PR TITLE
scripts to archive easyconfigs

### PR DESCRIPTION
**Pull request by Micket**
edit: wopps wrong base branch, forgot to swap to 5.0.x

### CODE REVIEW

File: scripts/archive_old [hunk 1]
```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python3
+
+import sys
```
Comment (smoors) [inline scripts/archive_old:3]: imported but not used

File: scripts/archive_toolchain [hunk 2]
```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env python3
+
+import sys
```
Comment (smoors) [inline scripts/archive_toolchain:3]: imported but not used

File: scripts/archive_toolchain [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+import git
+import glob
+import os
+import argparse
+
+
+parser = argparse.ArgumentParser(
+        prog='archive_old',
+        description='Archives easyconfigs based toolchain string')
```
Comment (smoors) [inline scripts/archive_toolchain:12]:
```suggestion
        description='Archives easyconfigs based on toolchain substring in easyconfig name')
```

File: scripts/archive_old [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+               return True
+    return False
+
+
+parser = argparse.ArgumentParser(
+        prog='archive_old',
+        description='Archives easyconfigs based on when they were first committed')
+parser.add_argument('--branch', action='store_true', help='create new branch')
+parser.add_argument('--apply', action='store_true', help='apply git mv')
+parser.add_argument('--repo', type=str, required=True)
```
Comment (smoors) [inline scripts/archive_old:24]: can you add help here?

File: scripts/archive_old [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+parser.add_argument(
+        '--cutoff',
+        type=lambda s: datetime.strptime(s, '%Y-%m-%d').astimezone(timezone.utc),
+        required=True)
```
Comment (smoors) [inline scripts/archive_old:28]: can you add help here too, with the date format and timezone to use?

**Comment by smoors** [comment]
would it be more useful if, when archiving a toolchain, say foss-2018b, all subtoolchains were also archived?
in our site we're keeping a mapping that can help with this.
can also be done in a follow-up PR of course.
```
SUBTOOLCHAINS = { 
    '2023a': ['GCCcore-12.3.0', 'GCC-12.3.0', 'intel-compilers-2023.1.0'],
    '2022b': ['GCCcore-12.2.0', 'GCC-12.2.0', 'intel-compilers-2022.2.1'],
    '2022a': ['GCCcore-11.3.0', 'GCC-11.3.0', 'intel-compilers-2022.1.0'],
    '2021b': ['GCCcore-11.2.0', 'GCC-11.2.0', 'intel-compilers-2021.4.0'],
    '2021a': ['GCCcore-10.3.0', 'GCC-10.3.0', 'intel-compilers-2021.2.0'],
    '2020b': ['GCCcore-10.2.0', 'GCC-10.2.0', 'iccifort-2020.4.304'],
    '2020a': ['GCCcore-9.3.0', 'GCC-9.3.0', 'iccifort-2020.1.217'],
    '2019b': ['GCCcore-8.3.0', 'GCC-8.3.0', 'iccifort-2019.5.281'],
    '2019a': ['GCCcore-8.2.0', 'GCC-8.2.0-2.31.1', 'iccifort-2019.1.144-GCC-8.2.0-2.31.1'],
    '2018b': ['GCCcore-7.3.0', 'GCC-7.3.0-2.30', 'iccifort-2018.3.222-GCC-7.3.0-2.30'],
}
```

**Comment by Micket** [comment]
I was requested to split it up and not make it one big PR

**Review by smoors — APPROVED** [review]
lgtm

**Pull request merged**